.github/workflows/ci.yml

```diff
@@ -29,26 +29,27 @@ jobs:
         run: npm ci
       - name: CI Node Engine Check
         run: npm run ci:checkNodeEngine
-  # check-lint:
-  #   name: Lint
-  #   timeout-minutes: 15
-  #   runs-on: ubuntu-18.04
-  #   steps:
-  #     - uses: actions/checkout@v2
-  #     - name: Use Node.js ${{ env.NODE_VERSION }}
-  #       uses: actions/setup-node@v1
-  #       with:
-  #         node-version: ${{ env.node-version }}
-  #     - name: Cache Node.js modules
-  #       uses: actions/cache@v2
-  #       with:
-  #         path: ~/.npm
-  #         key: ${{ runner.os }}-node-${{ env.NODE_VERSION }}-${{ hashFiles('**/package-lock.json') }}
-  #         restore-keys: |
-  #           ${{ runner.os }}-node-${{ env.NODE_VERSION }}-
-  #     - name: Install dependencies
-  #       run: npm ci
-  #     - run: npm run lint
+  check-jest:
+    name: Lint
+    timeout-minutes: 15
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ env.node-version }}
+      - name: Cache Node.js modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ env.NODE_VERSION }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ env.NODE_VERSION }}-
+      - name: Install dependencies
+        run: npm ci
+      - name: Test with Jest
+        run: npm test
```

Comment:
Why are we not using `npm run lint` for this? I think it would be easier, as we want to unify the linting and rules across repos and can use a shared config.

Reply:
What do you mean, adding name-run with `npm run lint`? The lint command is currently run in [pretest](https://github.com/RaschidJFR/parse-dashboard/blob/bugfix/fix_lint_and_tests/package.json#L130)

Reply:
I didn't notice that, it seems to be the only repo where lint runs in pretest. I suggest to remove pretest lint entirely, re-add the previously commented-out (now removed) lint CI job with `npm run lint` and maybe add the jtest test to the matrix job `check-build`, if it makes sense to run the tests under various node versions after building the bundles.

Reply:
I believe that it would make sense to run the e2e tests after the build has been created, although we [only have one](https://github.com/RaschidJFR/parse-dashboard/tree/bugfix/fix_lint_and_tests/src/lib/tests/e2e) for now.

But the rest of the tests are unit tests. From my short time spent in this repo, I perceive that they are loading components directly from the folder `src/`, so it doesn't make any difference to have the build created in the output folder. Although it wouldn't hurt to add this step in the job matrix, I think it makes more sense to add it during or before the build step.

And then the lining. We need to run it before Jest because unit tests fail if imports use double quotes instead of single quotes.

As I'm not very familiar with the github workflows, I'll trust your advice on where to add each of these 3 tasks (linting, unit tests, e2e).

Reply:
> loading components directly from the folder src/

I think we could improve that by loading from the built resource, to verify the build; we also do that in Parse Server repo tests where we load components from /lib instead of /src folder. That doesn't have to be part of this PR, but we can keep that in mind.

> And then the lining. We need to run it before Jest

The jobs are not run sequentially, they run in parallel as it is currently set up. Lint compliance should not have any code impact, just as lint-fixing should not have any code impact, it's merely a code style refactor.

> unit tests fail if imports use double quotes instead of single quotes.

That's strange where did you notice that?

> where to add each of these 3 tasks (linting, unit tests, e2e)

I'd suggest:
- re-add the commented out `check-lint` job where it originally was in the workflow file, just uncomment it and remove lint from `pretest` in `package.json`
- remove the `check-jest` job you added in this PR and just add a step to the existing `check-build` job:
  ```
   - name: Tests
      run: npm test
  ```